File: 20210523LangReview/Python/06-closure/closure.cs
using System;

namespace TS
{
    class TS 
    {
        public static void Main(string[] args)
        {
            Func<int> f = func(1);

            Console.WriteLine($"1st call, {0}", f());
            Console.WriteLine($"2nd call, {0}", f());
            Console.WriteLine($"3rd call, {0}", f());
        }

        static Func<int> func(int n)
        {
            int total = 0;
            int inner()
            {
                total += n;
                return total;
            }
            return inner;
        }
    }
}

File: 20210523LangReview/Python/06-closure/closure.js
/*
a simple module demonstrates closure mechnsim in JavaScript

@ZL, 20210525

[ expriment result ]
- by default, JavaScript has exactly same behavior (access local variable) like swift

*/

function func(n) {
    let total = 0;
    const f = ()=>{
        total += n;
        return total;
    }
    return f;
}

let f = func(10);

console.log("1st call, ", f());
console.log("2nd call, ", f());
console.log("3rd call, ", f());